sdk/core/core-client/src/urlHelpers.ts

```diff
@@ -218,15 +234,38 @@ function appendQueryParams(url: string, queryParams: Map<string, string | string
   // QUIRK: parsedUrl.searchParams will have their name/value pairs decoded, which
   // can change their meaning to the server, such as in the case of a SAS signature.
   // To avoid accidentally un-encoding a query param, we parse the key/values ourselves
-  const existingParams = simpleParseQueryParams(parsedUrl.search);
-  const combinedParams = new Map<string, string | string[]>(existingParams);
+  const combinedParams = simpleParseQueryParams(parsedUrl.search);
 
   for (const [name, value] of queryParams) {
     const existingValue = combinedParams.get(name);
     if (Array.isArray(existingValue)) {
-      existingValue.push(...value);
+      if (Array.isArray(value)) {
+        if (existingValue.toString() !== value.toString()) {
+          existingValue.push(...value);
+        }
+      } else {
+        existingValue.push(value);
+      }
     } else if (existingValue) {
-      combinedParams.set(name, [existingValue, ...value]);
+      if (operationSpec.queryParameters?.length) {
+        for (const queryParameter of operationSpec.queryParameters) {
+          if (queryParameter.mapper.serializedName === name) {
+            if (queryParameter.mapper.type.name === "Sequence") {
```

Comment:
This feels pretty expensive. Do we really have to hunt through all queryParams to see if it's a sequence or not in order to decide if we should replace the existing value? 

Could we maybe update `calculateQueryParameters` to return this information somehow? Like in addition to returning the map of queryParams, it could also return a Set of names that are every queryParam that is a sequence type? Then we can just check `sequenceParams.has(name)` and do the right thing?

Reply:
Done